tools/generate_leaderboard_plots.py

```diff
@@ -0,0 +1,235 @@
+}
+_tasks_names_map = {
+    "bach": "BACH",
+    "crc": "CRC",
+    "mhist": "MHIST",
+    "patch_camelyon": "PCam",
+    "camelyon16_small": "Cam16Small",
+    "panda_small": "PANDASmall",
+    "consep": "CoNSeP",
+    "monusac": "MoNuSAC",
+}
+_colors_for_startplot = [
+    "#7F7F7F",
+    "#FFC000",
+    "#C1E814",
+    "#FF0000",
+    "#D400FF",
+    "#4FFF87",
+    "#00A735",
+    "#6666FF",
+    "#0000FF",
+    "#00007F",
+    "#0000FF",
+]
+_label_offsets_startplot = {
+    "BACH": (0, -0.1),
+    "CRC": (0, -0.1),
+    "MHIST": (0.07, -0.1),
+    "PCam": (0.05, 0),
+    "Camelyon16": (-0.05, -0.08),
+    "PANDA": (0, -0.1),
+    "CoNSeP": (0.1, -0.07),
+    "MoNuSAC": (0.15, 0.08),
+}
+
+
+def get_leaderboard(logs_dir: Optional[str] = None) -> pd.DataFrame:
+    """Get the leaderboard data frame."""
+
+    # load existing leaderboard if available:
+    if os.path.isfile("tools/data/leaderboard.csv"):
+        df_existing = pd.read_csv("tools/data/leaderboard.csv")
+    else:
+        df_existing = pd.DataFrame()
+
+    # load results into data frame:
+    if logs_dir:
+        all_scores = []
+        for model in _fm_name_map.keys():
+            scores = []
+            for task in _tasks_to_metric.keys():
+                run_folder = [
+                    d
+                    for d in sorted(os.listdir(f"{logs_dir}/{task}/{model}/results"))
+                    if d.startswith("20")
+                ][-1]
+                with open(
+                    os.path.join(f"{logs_dir}/{task}/{model}/results/{run_folder}/results.json")
+                ) as f:
+                    d = json.load(f)
+                split = "test" if d["metrics"]["test"] else "val"
+                metric = _tasks_to_metric.get(task)
+                if metric is None:
+                    raise Exception(f"no metric defined for task {task}")
+                scores.append(round(d["metrics"][split][0][f"{split}/{metric}"]["mean"], 3))
+            all_scores.append(scores)
+        df = pd.DataFrame(all_scores, columns=_tasks_to_metric.keys())
+        df["model"] = _fm_name_map.keys()
+
+        # combine existing and new data frame
+        df = pd.concat([df, df_existing]).drop_duplicates()
+        df.to_csv("tools/data/leaderboard.csv", index=False)
+    else:
+        df = df_existing
+
+    df = df.set_index("model", drop=True)
+    df = df[[fm in _fm_name_map.keys() for fm in df.index]]
+    df.index = df.reset_index()["model"].apply(lambda x: _fm_name_map.get(x) or x)
+    return df
+
+
+def plot_leaderboard(df: pd.DataFrame, output_file: str = "docs/images/leaderboard.svg"):
+    """Plot the leaderboard heatmap."""
+
+    # create colormap:
+    colors = [[0, "white"], [1, "#0000ff"]]
+    cmap = matplotlib.colors.LinearSegmentedColormap.from_list("", colors)
+
+    # prepare data frame:
+    df.index.names = [""]
+
+    # exclude BACH from the average
+    df_for_avg = df[[c for c in df.columns if c != "bach"]]
```

Comment:
should we specify a list above `_exclude_from_avg = ["bach"]` to specify datasets to exclude?

Reply:
done